Closes #1808 - `ak.in1d()` Runtime Errors

Ethan-DeBandi99 opened the pull request:
Closes #1808 

This updates the JSON parsing to properly format JSON for server messages. The issue was that the types breaking were not in the dispatch table. However, since all of the types return true for `isinstance(obj, pdarray)`, The factory method has been updated to check if the value is a pdarray and then run the pdarray conversion.  This prevents the need for multiple entries in the dispatch table doing the same thing.

reuster986 reviewed (changes requested):
Code looks good! Could we add a test for `ak.in1d()` with some client dtype classes, just to be safe?

Ethan-DeBandi99 commented:
> Code looks good! Could we add a test for `ak.in1d()` with some client dtype classes, just to be safe?

@reuster986 - I added testing for the client dtype classes listed in the issue. I also added the json parsing to the message_test for these types to ensure that the message parameters are formatted correctly.

pierce314159 reviewed (approved):
Looks good!

reuster986 reviewed (approved):
excellent!

Ethan-DeBandi99 commented on tests/message_test.py:
```diff
@@ -145,9 +146,31 @@ def testJSONArgs(self):
+        # test ip_address
+        ip = ip_address(a)
+        size, args = _json_args_to_str({"ip": ip})
+        self.assertEqual(size, 1)
```
Good catch. I lost these lines and thought I had deleted. Removed and moved the `assertEqual` with the rest of the ip testing.
